src/gui/MainWindow.cc

```diff
@@ -3362,34 +3325,87 @@ void MainWindow::on_editActionFoldAll_triggered()
   activeEditor->foldUnfold();
 }
 
+QString MainWindow::getCurrentFileName() const
+{
+    if(activeEditor == nullptr)
+        return QString("");
+
+    const QFileInfo fileInfo(activeEditor->filepath);
+    QString fname = _("Untitled.scad");
+    if (!fileInfo.fileName().isEmpty()) fname = fileInfo.fileName();
+    return fname.replace("&", "&&");
+}
+
+void MainWindow::onTabManagerEditorChanged(EditorInterface* neweditor)
+{
+    activeEditor = neweditor;
+
+    if(neweditor == nullptr)
+        return;
+
+    parameterDock->setWidget(neweditor->parameterWidget);
+    editActionUndo->setEnabled(neweditor->canUndo());
+
+    QString name = getCurrentFileName();
+    setWindowTitle(name);
+
+    consoleDock->setNameSuffix(name);
+    errorLogDock->setNameSuffix(name);
+    animateDock->setNameSuffix(name);
+    fontListDock->setNameSuffix(name);
+    viewportControlDock->setNameSuffix(name);
+}
+
 void MainWindow::activateWindow(int offset)
 {
-  const std::array<DockFocus, 7> docks = {{
-    { editorDock, &MainWindow::on_windowActionSelectEditor_triggered },
-    { consoleDock, &MainWindow::on_windowActionSelectConsole_triggered },
-    { errorLogDock, &MainWindow::on_windowActionSelectErrorLog_triggered },
-    { parameterDock, &MainWindow::on_windowActionSelectCustomizer_triggered },
-    { fontListDock, &MainWindow::on_windowActionSelectFontList_triggered },
-    { animateDock, &MainWindow::on_windowActionSelectAnimate_triggered },
-    { viewportControlDock, &MainWindow::on_windowActionSelectViewportControl_triggered },
+  const std::array<Dock*, 7> docks = {{
+    editorDock,
+    consoleDock,
+    errorLogDock,
+    parameterDock,
+    fontListDock,
+    animateDock,
+    viewportControlDock,
```

Comment:
Not sure it's worth changing, but since we already have a vector of dock widgets elsewhere, it could be worth merging them, so they don't ever get out of sync.

Reply:
I agreed. I will do that.